### DISCUSSION TITLE
Allow import i18n-t tag as es module

### DISCUSSION THREAD

**Discussion by santiagoarizti** [kazupon/vue-i18n#1435] [Ideas]
(sorry for posting here, I created the same post in vue-18n-next [here](https://github.com/intlify/vue-i18n-next/discussions/872))

Hi, thanks for the awesome project.

I am using the v9 of vue-i18n with vue3

I am having issues with my IDE (phpstorm) recognizing the component `i18n-t` in my vue SFCs (single file components)

It would be of great help for the IDE (and therefore for the developer experience), to be able to import the component along the `useI18n` composable. This would greatly improve autocomplete and intellisense in the ide.

This is how I would import the component:

```
<script>
// the second import here is the component
import {useI18n, I18nT} from 'vue-i18n';

export default {
  components: {
    'i18n-t': I18nT, // note that in <script setup> this line is not necessary
  },
}
</script>
<template>
  <i18n-t keypath="hello">
    <strong>world</strong>
  </i18n-t>
</template>
```

This works fine today without need for the import, but my ide marks this as a warning 

![image](https://user-images.githubusercontent.com/8507324/148822007-24d5b60d-6130-4f8d-afa5-ff86c9afee5e.png)

I could add this tag as "ignored tags" in phpstorm

![image](https://user-images.githubusercontent.com/8507324/148821922-6d7ad634-e06a-4b9c-a4e8-4c9dad0e3f4c.png)

But I don't get any intellisense/help from the IDE about props for the tag `i18n-t`

What are your thoughts? is there any other way to make the IDE help me with this tag?

**Comment by dr-erych** [comment]
Just found out that the Translation component can be imported like this (it even needs to be done like that if using the composition API with `<script setup>`):

```vue
<template>
  <i18n-t keypath="term1" />
</template>

<script setup>
  import { useI18n, Translation as I18nT } from 'vue-i18n'
</script>
```

**Reply by szulcus** [reply]
I get this error:
```
Type '{ keypath: string; }' is not assignable to type 'ComponentProps<{ name: string; props: { keypath: { type: StringConstructor; required: boolean; }; plural: { type: (StringConstructor | NumberConstructor)[]; validator: (val: any) => boolean; }; } & { ...; }; setup(props: any, context: any): any; }>'.
  Type '{ keypath: string; }' is missing the following properties from type '{ name: string; props: { keypath: { type: StringConstructor; required: boolean; }; plural: { type: (StringConstructor | NumberConstructor)[]; validator: (val: any) => boolean; }; } & { ...; }; setup(props: any, context: any): any; }': name, props, setupts(2322)
```
![image](https://user-images.githubusercontent.com/48328739/196205753-56be881c-beed-494f-ba45-64b5af5fb88b.png)

What's wrong with that?

**Reply by dr-erych** [reply]
I ran into the same problem. The `Translation` object that is exported from `vue-i18n` needs to get passed to `defineComponent` so that it gets the intended type. My fix was creating a wrapper component:

src/components/I18n.ts:
```ts
import { defineComponent } from 'vue'
import { Translation } from 'vue-i18n'

export default defineComponent(Translation)
```

... and then using that one instead of directly importing from vue-i18n:
```vue
<template>
  <I18nT keypath="term1" />
</template>

<script setup>
  import { useI18n } from 'vue-i18n'
  import I18nT from '@/components/I18nT.ts'
</script>
```
